cli/ballerina-cli/src/main/java/io/ballerina/cli/task/RunTestsTask.java

```diff
@@ -228,14 +233,25 @@ public void execute(Project project) {
                     module.isDefaultModule() ? moduleName.toString() : module.moduleName().moduleNamePart();
             testSuiteMap.put(resolvedModuleName, suite);
             moduleNamesList.add(resolvedModuleName);
+            Map<String, String> mockFunctionMap = suite.getMockFunctionNamesMap();
+            for (Map.Entry<String, String> entry : mockFunctionMap.entrySet()) {
+                String key = entry.getKey();
+                String functionToMockClassName;
+                if (key.contains(MOCK_ANNOTATION_DELIMITER)) {
```

Comment:
Shall we check whether this logic fails if the function name contains `#` or `~`?

Reply:
I have used the same logic in other usages as well. Seems not working. I have created a issue since same issue can be observed update 1 as well.

Reply:
Created https://github.com/ballerina-platform/ballerina-lang/issues/37027